validation.go

```diff
@@ -74,6 +78,14 @@ func validateJoin(ctx *validation.Context) bool {
 	return true
 }
 
+func validateSearch(ctx *validation.Context) bool {
```

Comment:
Because "search" is only a string, it's not necessary. We can just use the built-in `string` rule.